### PR TITLE
[LM-9706] Shimmer in Home Feed and DMFeed and few remaining keys flow

### PR DESCRIPTION
**Pull request by arnavr15**
## Description

**Features**
- Added shimmer in HomeFeed and DMFeed
- Added few remaining keys flow
- Added chatroom follow/unfollow flow

**Bugs**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (fixes a breaking changes)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Other, please describe:

### Author Checklist

- [x] Assignee added
- [x] Reviewer added
- [x] Description added
- [ ] Unit test cases added

### Reviewer Checklist

- [x] PR title starts with Jira ticket number prefix
- [ ] If hotfix, Add Hotfix as prefix in PR title
- [x] Label added
- [x] Commit messages follows standard format (added, changed, removed, code commented and chore)
- [x] Description about the Feature/ Bugs added

### CODE REVIEW

File: ChatSX/screens/ChatRoom/index.tsx [hunk 1]
```diff
@@ -118,10 +121,9 @@ import {CognitoIdentityCredentials, S3} from 'aws-sdk';
 import AWS from 'aws-sdk';
 import {FlashList} from '@shopify/flash-list';
 import WarningMessageModal from '../../customModals/WarningMessage';
+import AsyncStorage from '@react-native-async-storage/async-storage';
```
Comment (ishaan1607) [inline ChatSX/screens/ChatRoom/index.tsx:124]: remove this

File: ChatSX/screens/ChatRoom/index.tsx [hunk 2]
```diff
@@ -956,131 +960,39 @@ const ChatRoom = ({navigation, route}: ChatRoom) => {
...
-  const handleOnScroll: ScrollViewProps['onScroll'] = event => {
-    const offset = event.nativeEvent.contentOffset.y;
-    const visibleLength = event.nativeEvent.layoutMeasurement.height;
-    const contentLength = event.nativeEvent.contentSize.height;
-    const onStartReachedThreshold = 10;
-    const onEndReachedThreshold = 10;
-
-    // Check if scroll has reached start of list.
-    const isScrollAtStart = offset < onStartReachedThreshold;
-    // Check if scroll has reached end of list.
```
Comment (ishaan1607) [inline ChatSX/screens/ChatRoom/index.tsx:1055]: revert this change

File: ChatSX/screens/ChatRoom/index.tsx [hunk 3]
```diff
@@ -1200,8 +1112,10 @@ const ChatRoom = ({navigation, route}: ChatRoom) => {
           };
           await dispatch(getExploreFeedData(payload2, true) as any);
           updatePageInRedux();
+          // await dispatch(getHomeFeedData({page: 1}) as any);
         } else {
           updatePageInRedux();
+          // await dispatch(getHomeFeedData({page: 1}) as any);
```
Comment (ishaan1607) [inline ChatSX/screens/ChatRoom/index.tsx:1118]: remove

File: ChatSX/screens/ChatRoom/index.tsx [hunk 4]
```diff
@@ -1246,8 +1161,10 @@ const ChatRoom = ({navigation, route}: ChatRoom) => {
           };
           await dispatch(getExploreFeedData(payload2, true) as any);
           updatePageInRedux();
+          // await dispatch(getHomeFeedData({page: 1}) as any);
```
Comment (ishaan1607) [inline ChatSX/screens/ChatRoom/index.tsx:1164]: remove

File: ChatSX/screens/ChatRoom/index.tsx [hunk 5]
```diff
@@ -1246,8 +1161,10 @@ const ChatRoom = ({navigation, route}: ChatRoom) => {
...
         } else {
           updatePageInRedux();
+          // await dispatch(getHomeFeedData({page: 1}) as any);
```
Comment (ishaan1607) [inline ChatSX/screens/ChatRoom/index.tsx:1167]: remove

**Review by ishaan1607 — CHANGES_REQUESTED** [review]

File: ChatSX/screens/ChatRoom/index.tsx [hunk 6]
```diff
@@ -618,78 +606,85 @@ const ChatRoom = ({navigation, route}: ChatRoom) => {
...
   ) {
     const res = await myClient?.syncConversation(
       SyncConversationRequest.builder()
         .setChatroomId(chatroomID)
         .setPage(page)
         .setMinTimestamp(minTimeStamp)
         .setMaxTimestamp(maxTimeStamp)
-        .setPageSize(20)
+        .setPageSize(500)
+        .setIsLocalDb(!!conversationId ? false : true)
```
[the remark this answers is not in the document]
Comment (arnavr15) [reply ChatSX/screens/ChatRoom/index.tsx:618]: will remove it

**Review by ishaan1607 — APPROVED** [review]
